### PR TITLE
server: fix to start span before logging

### LINKED ISSUE #1976: server: logged traces are always empty
### Describe the bug

Server always logs empty traces (i.e., `"logging.googleapis.com/trace":"00000000000000000000000000000000"`).

### To Reproduce

Run server with logging on.

```
server.New(handler, &server.Options{
	RequestLogger: requestlog.NewStackdriverLogger(os.Stdout, nil),
})
```

## Expected behavior

It should output non-empty traces (e.g., `"logging.googleapis.com/trace":"ed9828b4ed67b6600c1e0c2ba1782f01"`).

### Version

`v0.13.0`.

### Additional context

None.

### PR DESCRIPTION
**Pull request by shouichi**
Fixes #1976.

### CODE REVIEW

**Comment by shouichi** [comment]
> Would you be willing to write a unit test for this change?

Sure 👍

**Comment by shouichi** [comment]
@zombiezen Added a test!

File: server/server_test.go [hunk 1]
```diff
@@ -18,10 +18,13 @@ import (
...
 	"errors"
 	"net/http"
+	"net/http/httptest"
 	"testing"
+
+	"gocloud.dev/requestlog"
 )
 
-func TestListenAndServe(t *testing.T) {
+func TestDriver(t *testing.T) {
```
Comment (zombiezen) [inline server/server_test.go]: Let's keep this `ListenAndServe`, because that is what this test is verifying.

File: server/server_test.go [hunk 2]
```diff
@@ -33,15 +36,53 @@ func TestListenAndServe(t *testing.T) {
 	}
```
Comment (zombiezen) [inline server/server_test.go]: Let's also check that `td.handler != nil`, since we depend on that in the new test.

File: server/server_test.go [hunk 3]
```diff
@@ -33,15 +36,53 @@ func TestListenAndServe(t *testing.T) {
...
 }
 
+func TestLogging(t *testing.T) {
```
Comment (zombiezen) [inline server/server_test.go]:
```suggestion
func TestMiddleware(t *testing.T) {
```

(We may want to verify other properties of the middleware later.)

File: server/server_test.go [hunk 4]
```diff
@@ -33,15 +36,53 @@ func TestListenAndServe(t *testing.T) {
...
+		t.Fatal(err)
+	}
+
+	req, err := http.NewRequest("GET", "/", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rr := httptest.NewRecorder()
+	td.handler.ServeHTTP(rr, req)
```
Comment (zombiezen) [inline server/server_test.go]: We should also verify that `onLog` ends up being called.

**Comment by shouichi** [comment]
@zombiezen Thanks for the feedback, fixed the test 🚀

**Pull request merged**